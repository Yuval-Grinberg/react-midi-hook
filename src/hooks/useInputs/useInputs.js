import { useEffect, useState } from 'react'

export default function useInputs() {
  const [inputs, setInputs] = useState(null)

  useEffect(() => {
    navigator.requestMIDIAccess({ sysex: true }).then(
      (event) => setInputs(event.inputs.values()),
      (event) => console.warn('error')
    )
  }, [setInputs])

  return inputs
}
