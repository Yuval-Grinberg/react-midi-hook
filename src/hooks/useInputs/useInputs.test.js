import { renderHook } from '@testing-library/react-hooks';
import useInputs from './useInputs';

const mockMIDIAccess = jest.fn().mockImplementation(() => {
  return Promise.resolve({
    inputs: new Set([{ id: '0', name: 'Testing', onmidimessage: '' }])
  });
});

describe('useInputs', () => {
  describe('useEffect', () => {
    beforeAll(() => {
      global.navigator.requestMIDIAccess = mockMIDIAccess;
    });

    it('sets inputs once loaded', async () => {
      const { result, waitForNextUpdate } = renderHook(useInputs);
      expect(result.current.inputs).toMatchSnapshot();
      await waitForNextUpdate();
      expect(result.current.inputs).toMatchSnapshot();
    });
  });
});
