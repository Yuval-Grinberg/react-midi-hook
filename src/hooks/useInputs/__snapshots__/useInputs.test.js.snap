// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useInputs useEffect sets inputs once loaded 1`] = `Array []`;

exports[`useInputs useEffect sets inputs once loaded 2`] = `
Array [
  Object {
    "id": "0",
    "name": "Testing",
    "onmidimessage": undefined,
  },
]
`;
